### PR TITLE
NOISSUE - Fix dates not being init properly on save, change path construction, replace UUID with ULID for group ID

### PR DESCRIPTION
**Pull request by mteodor**
### What does this do?

* Fix `created_at` and `updated_at` not being saved properly
* Change the way that path is constructed, instead of the name of the group being used group id will be used
```
things=# select name, id, path from thing_groups;
  name   |             id             |                         path                          
---------+----------------------------+-------------------------------------------------------
 group13 | 01ESM3GW5VD6VSQ2YYDPCDQHXH | 01ESM3GW5VD6VSQ2YYDPCDQHXH
 group13 | 01ESM3H75DCYQPWHYZS8BAT733 | 01ESM3GW5VD6VSQ2YYDPCDQHXH.01ESM3H75DCYQPWHYZS8BAT733
```
* Name of the group can be changed
* Name of the group is not unique

### CODE REVIEW

**Review by drasko — CHANGES_REQUESTED** [review]
This looks a bot complex... let's think if it is really needed.

Also - does name still stays unique? We should not demand that it is unique if we use IDs.

File: things/postgres/groups.go [hunk 1]
```diff
@@ -574,21 +551,17 @@ func toGroup(dbu dbGroup) (groups.Group, error) {
 }
 
 type dbGroupRelation struct {
-	Group uuid.UUID `db:"group_id"`
+	Group string    `db:"group_id"`
 	Thing uuid.UUID `db:"thing_id"`
```
Comment (manuio) [inline things/postgres/groups.go]: This should be `GroupID` and `ThingID`

File: pkg/ulid/ulid.go [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+	"github.com/mainflux/mainflux/pkg/errors"
+	"github.com/oklog/ulid/v2"
+
+	mathrand "math/rand"
+)
+
+// ErrGeneratingID indicates error in generating ULID
+var ErrGeneratingID = errors.New("generating id failed")
+
+var _ mainflux.UUIDProvider = (*ulidProvider)(nil)
```
Comment (manuio) [inline pkg/ulid/ulid.go]: Should we rename this  to `mainflux.IDProvider` ?

File: things/service.go [hunk 3]
```diff
@@ -10,7 +10,7 @@ import (
 	"github.com/mainflux/mainflux/pkg/errors"
 
 	"github.com/mainflux/mainflux"
-	uuidProvider "github.com/mainflux/mainflux/pkg/uuid"
+	mfxulid "github.com/mainflux/mainflux/pkg/ulid"
```
Comment (drasko) [inline things/service.go]: Why `mfxulid` and not `ulidProvider` or `IDProvider`? Maybe we can also remove everything and use just `ulid`?

**Review by drasko — CHANGES_REQUESTED** [review]

**Review by drasko — DISMISSED** [review]
LGTM

**Review by manuio — DISMISSED** [review]
LGTM

File: internal/groups/api/requests.go [hunk 4]
```diff
@@ -50,10 +50,6 @@ func (req updateGroupReq) validate() error {
 		return groups.ErrParentInvariant
 	}
 
-	if req.Name != "" {
-		return groups.ErrNameInvariant
```
Comment (dborovcanin) [inline internal/groups/api/requests.go:54]: We should remove this error if it's not used elsewhere.

**Review by dborovcanin — APPROVED** [review]

**Review by manuio — APPROVED** [review]
LGTM

**Review by drasko — APPROVED** [review]
LGTM

**Pull request merged**